src/components/Modals/PublishCollectionModal/PublishCollectionModal.tsx

```diff
@@ -1,39 +1,120 @@
 import * as React from 'react'
-import { ModalNavigation, Button } from 'decentraland-ui'
+import { ModalNavigation, Button, Mana, Loader } from 'decentraland-ui'
 import Modal from 'decentraland-dapps/dist/containers/Modal'
 import { t } from 'decentraland-dapps/dist/modules/translation/utils'
 
-import { Props } from './PublishCollectionModal.types'
+import { builder } from 'lib/api/builder'
+import { fromWei } from 'web3x-es/utils'
+import { RARITY_COLOR, ItemRarity } from 'modules/item/types'
+import { Props, State } from './PublishCollectionModal.types'
 import './PublishCollectionModal.css'
 
-export default class PublishCollectionModal extends React.PureComponent<Props> {
-  componentDidMount() {
+export default class PublishCollectionModal extends React.PureComponent<Props, State> {
+  state: State = { step: 1, rarities: [], isFetchingRarities: true }
+
+  async componentDidMount() {
     const { collection, onClose } = this.props
     if (!collection) {
       onClose()
     }
+
+    const rarities = await builder.fetchRarities()
+    this.setState({ rarities, isFetchingRarities: false })
   }
 
   handlePublish = () => {
     const { collection, items, onPublish } = this.props
     onPublish(collection!, items)
   }
 
+  handleProceed = () => {
+    this.setState({ step: 2 })
+  }
+
+  renderFirstStep = () => {
+    const { items } = this.props
+    const { rarities, isFetchingRarities } = this.state
+
+    const itemsByRarity: Record<string, { name: ItemRarity; count: number; price: number }> = {}
+    let totalPrice = 0
+
+    if (rarities.length > 0) {
+      items.forEach(item => {
+        const rarity = rarities.find(r => r.name === item.rarity)!
```

Comment:
Two things, non important:
- Most of the codebase uses `for..of` for these kinds of things, leaving forEach for renders (where for...of does not work) or really particular cases. **It's the same thing** here, as you don't need to break early, **but** if instead of using `!` we'd like to skip the rarity, on a for...of we could do `if(!rarity) continue`. No need to change it
- Can we call `r` -> `rarity`? it's not that many more caracters but's a bit more clear